rest-api/alembic/versions/b4f6eb55d503_add_sample_and_identifier_history.py

```diff
@@ -0,0 +1,78 @@
+"""add sample and identifier history
+
+Revision ID: b4f6eb55d503
+Revises: 4aa79d5ac86e
+Create Date: 2018-08-31 13:26:22.482944
+
+"""
+from alembic import op
+import sqlalchemy as sa
+import model.utils
+
+
+from participant_enums import PhysicalMeasurementsStatus, QuestionnaireStatus, OrderStatus
+from participant_enums import WithdrawalStatus, SuspensionStatus, QuestionnaireDefinitionStatus
+from participant_enums import EnrollmentStatus, Race, SampleStatus, OrganizationType, BiobankOrderStatus
+from participant_enums import MetricSetType, MetricsKey
+from model.site_enums import SiteStatus, EnrollingStatus, DigitalSchedulingStatus, ObsoleteStatus
+from model.code import CodeType
+
+# revision identifiers, used by Alembic.
+revision = 'b4f6eb55d503'
+down_revision = '4aa79d5ac86e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('biobank_order_identifier_history',
+    sa.Column('system', sa.String(length=80), nullable=False),
+    sa.Column('value', sa.String(length=80), nullable=False),
+    sa.Column('version', sa.Integer(), nullable=False),
+    sa.Column('biobank_order_id', sa.String(length=80), nullable=False),
+    sa.ForeignKeyConstraint(['biobank_order_id'], ['biobank_order.biobank_order_id'], ),
+    sa.PrimaryKeyConstraint('system', 'value', 'version')
+    )
+    op.create_table('biobank_ordered_sample_history',
+    sa.Column('test', sa.String(length=80), nullable=False),
+    sa.Column('description', sa.UnicodeText(), nullable=False),
+    sa.Column('processing_required', sa.Boolean(), nullable=False),
+    sa.Column('collected', model.utils.UTCDateTime(), nullable=True),
+    sa.Column('processed', model.utils.UTCDateTime(), nullable=True),
+    sa.Column('finalized', model.utils.UTCDateTime(), nullable=True),
+    sa.Column('version', sa.Integer(), nullable=False),
+    sa.Column('order_id', sa.String(length=80), nullable=False),
+    sa.ForeignKeyConstraint(['order_id'], ['biobank_order.biobank_order_id'], ),
+    sa.PrimaryKeyConstraint('test', 'version', 'order_id')
```

Comment:
This should be order_id, test, version, or order_id, version, test... Putting order_id first matters if you want to make scanning the samples for an order efficient.

It looks like you can get SQLAlchemy to do this properly using __table_args__, instead of primary_key=True, as described here:

https://stackoverflow.com/questions/9034271/sqlalchemy-orm-how-to-declare-a-table-class-that-contains-multi-column-primary